utils/highway-rewards-analysis/src/main.rs

```diff
@@ -0,0 +1,201 @@
+    if state.faulty_validators().count() == 0 {
+        return;
+    }
+
+    println!("Faults:");
+    for vid in state.faulty_validators() {
+        let fault = state.maybe_fault(vid).unwrap();
+        println!("{}: {:?}", validators.id(vid).unwrap(), fault);
+    }
+    println!();
+}
+
+const TOP_TO_PRINT: usize = 10;
+
+fn round_num(dump: &EraDump, round_id: Timestamp) -> u64 {
+    let min_round_length = dump.highway_state.params().min_round_length();
+    (round_id.millis() - dump.start_time.millis()) / min_round_length.millis()
+}
+
+fn print_skipped_rounds(validators: &Validators<PublicKey>, dump: &EraDump, verbose: bool) {
+    let state = &dump.highway_state;
+    let highest_block = state.fork_choice(state.panorama()).unwrap();
+    let all_blocks = std::iter::once(highest_block).chain(state.ancestor_hashes(highest_block));
+    let mut skipped_rounds = vec![vec![]; validators.len()];
+
+    for block_hash in all_blocks {
+        let block_unit = state.unit(block_hash);
+        let round_id = block_unit.round_id();
+        for i in 0..validators.len() as u32 {
+            let observation = state.panorama().get(i.into()).unwrap();
+            let round_participation = round_participation(state, observation, round_id);
+            if matches!(round_participation, RoundParticipation::No) {
+                skipped_rounds[i as usize].push(round_id);
+            }
+        }
+    }
+
+    for rounds in skipped_rounds.iter_mut() {
+        rounds.sort();
+    }
+
+    let mut num_skipped_rounds: Vec<_> = skipped_rounds
+        .iter()
+        .enumerate()
+        .map(|(i, skipped)| (i as u32, skipped.len()))
+        .collect();
+    num_skipped_rounds.sort_by_key(|(_, len)| *len);
+
+    println!("{} validators who skipped the most rounds:", TOP_TO_PRINT);
+    for (vid, len) in num_skipped_rounds.iter().rev().take(TOP_TO_PRINT) {
+        println!(
+            "{}: skipped {} rounds",
+            validators.id((*vid).into()).unwrap(),
+            len
+        );
+    }
+
+    if verbose {
+        println!();
+        for (vid, _) in num_skipped_rounds.iter().rev().take(TOP_TO_PRINT) {
+            let skipped_rounds: Vec<_> = skipped_rounds[*vid as usize]
+                .iter()
+                .map(|rid| format!("{}", round_num(dump, *rid)))
+                .collect();
+            println!(
+                "{} skipped rounds: {}",
+                validators.id((*vid).into()).unwrap(),
+                skipped_rounds.join(", ")
+            );
+        }
+    }
+
+    println!();
+}
+
+fn print_lowest_quorum_participation(validators: &Validators<PublicKey>, dump: &EraDump) {
+    let state = &dump.highway_state;
+    let highest_block = state.fork_choice(state.panorama()).unwrap();
+    let mut quora_sum = vec![0.0; validators.len()];
+    let mut num_rounds = 0;
+
+    let hb_unit = state.unit(highest_block);
+    for bhash in state.ancestor_hashes(highest_block) {
```

Comment:
Some comments would be useful within this loop, to avoid having to go consult the paper every time